flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterServiceLeadershipRunnerTest.java

```diff
@@ -662,15 +663,21 @@ public void testJobMasterServiceProcessCreationFailureIsForwardedToResultFuture(
 
     @Test
     public void testJobAlreadyDone() throws Exception {
-        JobID jobID = new JobID();
+        JobID jobId = new JobID();
+        JobResult jobResult =
+                new JobResult.Builder()
+                        .applicationStatus(ApplicationStatus.UNKNOWN)
```

Comment:
We might not want to use a state that shouldn't appear in production as already pointed out in the `EmbeddedJobResultStore`.